addon/-private/system/relationships/state/create.js

```diff
@@ -15,6 +15,8 @@ function createRelationshipFor(record, relationshipMeta, store) {
   let inverse = null;
   if (shouldFindInverse(relationshipMeta)) {
     inverse = record.type.inverseFor(relationshipMeta.key, store);
+  } else {
+    record.type.typeForRelationship(relationshipMeta.key, store);
```

Comment:
I think the correct code here to ensure we throw when the inverse is an undefined model is:

```
import { runInDebug } from 'ember-data/-private/system/debug';

...

runInDebug(() => {
  // for perf reasons, this should be lazy, but for explicit `inverse: null;` situations
  // we need to run the lookup to throw the assert if it were improperly defined (e.g. the related model is not defined).
  record.type.typeForRelationship(relationshipMeta.key, store);
});
```